src/restrictededitingmodeediting.js

```diff
@@ -311,34 +309,31 @@ function getCommandExecuter( editor, commandName ) {
 	};
 }
 
-// Helper for handling custom keydown behaviour.
-export function onKeyDown( editor, eventInfo, evtData ) {
+// Helper for handling Ctrl+A keydown behaviour.
+function onSelectAll( editor, data, cancel ) {
 	const model = editor.model;
-	const selection = model.document.selection;
-	const ctrlA = evtData.ctrlKey && evtData.keyCode === 65;
+	const selection = editor.model.document.selection;
+	const marker = getMarkerAtPosition( editor, selection.focus );
+
+	if ( !marker ) {
+		return;
+	}
 
-	// Ctrl+A handler.
 	// If selection range is inside a restricted editing exception, select text only within the exception.
 	//
-	// Note: Second Ctrl+A will select the entire text in the editor.
-	if ( ctrlA ) {
-		const marker = getMarkerAtPosition( editor, selection.focus );
-
-		if ( marker ) {
-			const { start: { offset: selecitonStart }, end: { offset: selecitonEnd } } = selection.getFirstRange();
-			const { start: { offset: markerStart }, end: { offset: markerEnd } } = marker.getRange();
+	// Note: Second Ctrl+A press is also blocked and it won't select the entire text in the editor.
+	// Just like in the widget.
+	const selectionRange = selection.getFirstRange();
+	const markerRange = marker.getRange();
 
-			const selectionIsInsideException = selecitonStart > markerStart && selecitonEnd < markerEnd;
+	const selectionIsInsideException = markerRange.containsRange( selectionRange, true );
```

Comment:
Might be inlined to `if` - this check isn't that complicated to be outside if.

Alternatively, you could make whole if statement as `shouldSelectWholeException` (or similar) but that is not needed IMO.